OpenRA.Mods.Common/Widgets/Logic/Editor/ActorSelectorLogic.cs

```diff
@@ -77,25 +97,119 @@ public ActorSelectorLogic(Widget widget, World world, WorldRenderer worldRendere
 			ownersDropDown.Text = selectedOwner.Name;
 			ownersDropDown.TextColor = selectedOwner.Color.RGB;
 
-			var actorCategorySelector = widget.Get<DropDownButtonWidget>("ACTOR_CATEGORY");
-			var filtersPanel = Ui.LoadWidget("ACTOR_CATEGORY_FILTER_PANEL", null, new WidgetArgs());
-			var categoryTemplate = filtersPanel.Get<CheckboxWidget>("CATEGORY_TEMPLATE");
 			var tileSetId = world.Map.Rules.TileSet.Id;
-			allCategories = mapRules.Actors.Where(a => !a.Value.Name.Contains('^')).Select(a => a.Value.TraitInfoOrDefault<EditorTilesetFilterInfo>())
-				.Where(i => i != null && i.Categories != null &&
-					!(i.ExcludeTilesets != null && i.ExcludeTilesets.Contains(tileSetId)) && !(i.RequireTilesets != null && !i.RequireTilesets.Contains(tileSetId)))
-				.SelectMany(i => i.Categories).Distinct().OrderBy(i => i).ToList();
-			selectedCategories.AddRange(allCategories);
+			var allActorsTemp = new List<ActorSelectorActor>();
+			foreach (var a in mapRules.Actors.Values)
+			{
+				// Partial templates are not allowed
+				if (a.Name.Contains('^'))
+					continue;
+
+				// Actor must have a preview associated with it
+				if (!a.HasTraitInfo<IRenderActorPreviewInfo>())
+					continue;
+
+				var editorData = a.TraitInfoOrDefault<EditorTilesetFilterInfo>();
+
+				// Actor must be included in at least one category
+				if (editorData == null || editorData.Categories == null)
+					continue;
+
+				// Excluded by tileset
+				if (editorData.ExcludeTilesets != null && editorData.ExcludeTilesets.Contains(tileSetId))
+					continue;
+
+				if (editorData.RequireTilesets != null && !editorData.RequireTilesets.Contains(tileSetId))
+					continue;
+
+				var tooltip = a.TraitInfos<EditorOnlyTooltipInfo>().FirstOrDefault(Exts.IsTraitEnabled) as TooltipInfoBase
+					?? a.TraitInfos<TooltipInfo>().FirstOrDefault(Exts.IsTraitEnabled);
```

Comment:
`a.TraitInfos<EditorOnlyTooltipInfo>().FirstOrDefault()` (since it's bogus for editor-only tooltips to be disabled by default) and `a.TraitInfos<TooltipInfo>().FirstOrDefault(i => i.EnabledByDefault)` should be correct here.